src/ac_power_flow_jacobian.jl

```diff
@@ -1,228 +1,511 @@
-                    push!(J0_I, F_ix_f_r)
-                    push!(J0_J, X_ix_t_fst)
-                    push!(J0_V, 0.0)
-                    #Rective PF w/r Local Reactive Power
-                    push!(J0_I, F_ix_f_i)
-                    push!(J0_J, X_ix_t_snd)
-                    push!(J0_V, 0.0)
-                end
-            elseif nb == PSY.ACBusTypes.PV
-                #Active PF w/r Angle
-                push!(J0_I, F_ix_f_r)
-                push!(J0_J, X_ix_t_snd)
-                push!(J0_V, 0.0)
-                #Reactive PF w/r Angle
-                push!(J0_I, F_ix_f_i)
-                push!(J0_J, X_ix_t_snd)
-                push!(J0_V, 0.0)
-                if ix_f == ix_t
-                    #Reactive PF w/r Local Reactive Power
-                    push!(J0_I, F_ix_f_i)
-                    push!(J0_J, X_ix_t_fst)
-                    push!(J0_V, 0.0)
-                end
-            elseif nb == PSY.ACBusTypes.PQ
-                #Active PF w/r VoltageMag
-                push!(J0_I, F_ix_f_r)
-                push!(J0_J, X_ix_t_fst)
-                push!(J0_V, 0.0)
-                #Reactive PF w/r VoltageMag
-                push!(J0_I, F_ix_f_i)
-                push!(J0_J, X_ix_t_fst)
-                push!(J0_V, 0.0)
-                #Active PF w/r Angle
-                push!(J0_I, F_ix_f_r)
-                push!(J0_J, X_ix_t_snd)
-                push!(J0_V, 0.0)
-                #Reactive PF w/r Angle
-                push!(J0_I, F_ix_f_i)
-                push!(J0_J, X_ix_t_snd)
-                push!(J0_V, 0.0)
-            end
+    for bus_from in 1:num_buses
+        row_from_p = 2 * bus_from - 1  # Row index for the value that is related to active power
+        row_from_q = 2 * bus_from      # Row index for the value that is related to reactive power
+        for bus_to in data.neighbors[bus_from]
+            col_to_vm = 2 * bus_to - 1  # Column index for the value related to voltage magnitude
+            col_to_va = 2 * bus_to      # Column index for the value related to voltage angle
+            # We ignore the bus type and initialize the structure as if all buses were PQ - 
+            # mainly because we can have a PV -> PQ transition, and the number of REF buses is small
+            # bus_type = data.bus_type[bus_to, time_step]
+            _create_jacobian_matrix_structure_bus!(
+                rows,
+                columns,
+                values,
+                bus_from,
+                bus_to,
+                row_from_p,
+                row_from_q,
+                col_to_vm,
+                col_to_va,
+                # Val(bus_type),
+            )
         end
     end
-    J0 = SparseArrays.sparse(J0_I, J0_J, J0_V)
-    return J0
+    Jv0 = SparseArrays.sparse(rows, columns, values)
+    return Jv0
+end
+
+function _set_entries_for_neighbor(Jv::SparseArrays.SparseMatrixCSC{Float64, Int32},
+    Yb::SparseArrays.SparseMatrixCSC{ComplexF64, Int},
+    Vm::SubArray{
```

Comment:
Do we really need to type this so specifically? What is the benefit to doing so?

Reply:
replaced by AbstractArray{Float64},